waiter/integration/waiter/basic_test.clj

```diff
@@ -53,6 +54,35 @@
           (is (= "0" (get-in body-json ["headers" "content-length"])) (str body))
           (is (= "text/plain" (get-in body-json ["headers" "accept"])) (str body))))
 
+      (testing "query-string with special characters"
+        (log/info "Basic test for query-string with special characters")
+        (let [{:keys [body] :as response} (make-kitchen-request
```

Comment:
`jet` encodes the query parameters, so we do not see the errors from this testing clause. Hence, we need to use the raw `HttpURLConnection` version below to create the issue.